### PR TITLE
Add support for parsing PDF pages in parallel (multiprocessing)

### PR DESCRIPTION
**Pull request by phoewass**
Closes https://github.com/py-pdf/pypdf_table_extraction/discussions/8

Parse pages in parallel using multiprocessing library leveraging all the available CPUs.

Checklist:
- [x] Process in parallel using the library
- [x] Process in parallel using the CLI
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: camelot/handlers.py [hunk 1]
```diff
@@ -171,14 +179,54 @@ def parse(
...
+            cpu_count = mp.cpu_count()
+            # Using multiprocessing only when cpu_count > 1 to prevent a stallness issue
+            # when cpu_count is 1
+            if parallel and len(self.pages) > 1 and cpu_count > 1:
+                with mp.get_context("spawn").Pool(processes=cpu_count) as pool:
+                    jobs = [
+                        pool.apply_async(
+                            self._parse_page,
+                            (p, tempdir, parser, suppress_stdout, layout_kwargs)
+                        ) for p in self.pages
```
Comment (foarsitter) [inline camelot/handlers.py:193]: This list comprehension makes the code less readable. 

Also I would opt to move the code inside a `parse_parallel` method.

Comment (phoewass) [reply camelot/handlers.py]: I moved it to a simple loop.

OTOH I would keep the parallel and sequential processing close to each other to be clear at a glance that the arguments are the same:
```python
j = pool.apply_async(self._parse_page,(p, tempdir, parser, suppress_stdout, layout_kwargs))
```
and
```python
t = self._parse_page(p, tempdir, parser, suppress_stdout, layout_kwargs)
```

Comment (foarsitter) [reply camelot/handlers.py]: Good thinking! Thanks for your clarification.

**Review by foarsitter — APPROVED** [review]

**Review by bosd — APPROVED** [review]

**Comment by bosd** [comment]
@foarsitter Should we go ahead and merge this?

**Pull request merged**